src/multi_vector_simulator/C1_verification.py

```diff
@@ -766,3 +770,72 @@ def check_for_sufficient_assets_on_busses(dict_values):
                 f"so that the bus is not a dead end should be two, excluding the excess sink. "
                 f"These are the connected assets: {asset_string}"
             )
+
+
+def check_energy_system_can_fulfill_max_demand(dict_values):
+    r"""
+    Logs a logging.warning message if the aggregated installed capacity and maximum capacity (if applicable)
+    of all conversion, generation and storage assets connected to one bus is smaller than the maximum demand.
+    
+   The check is applied to each bus of the energy system. Check passes when the potential peak supply is
+   larger then or equal to the peak demand on the bus, or if the maximum capacity of an asset is set to
+   None for optimization.
+
+    Parameters
+    ----------
+    dict_values : dict
+        Contains all input data of the simulation.
+
+    Returns
+    -------
+    Indirectly, logs a logging.warning message if the installed and maximum capacities of
+    conversion/generation/storage assets are less than the maximum demand, for each bus.
+
+    Notes
+    -----
+    Tested with:
+    - test_check_energy_system_can_fulfill_max_demand_sufficient_capacities()
+    - test_check_energy_system_can_fulfill_max_demand_no_maximum_capacity()
+    - test_check_energy_system_can_fulfill_max_demand_insufficient_capacities()
+    - test_check_energy_system_can_fulfill_max_demand_with_storage()
+    """
+    for bus in dict_values[ENERGY_BUSSES]:
+        skip_check = False
+        opt_cap_storage = False
+        peak_demand = 0
+        peak_generation = 0
+        for item in dict_values[ENERGY_BUSSES][bus][ASSET_DICT]:
+            # filters out excess energy sinks, leaving only actual demand profiles
+            if item in dict_values[ENERGY_CONSUMPTION] and \
+                    dict_values[ENERGY_CONSUMPTION][item][DISPATCHABILITY][VALUE] is False:
+                peak_demand += dict_values[ENERGY_CONSUMPTION][item][TIMESERIES_PEAK][VALUE]
+            for group in [ENERGY_CONVERSION, ENERGY_PRODUCTION]:
+                if item in dict_values[group]:
+                    peak_generation += dict_values[group][item][INSTALLED_CAP][VALUE]
```

Comment:
I just noticed that, connected to #825, for PV assets the maximum output is actually not the `INSTALLED_CAP`, but `INSTALLED_CAP*timeseries peak`, same for `MAXIMUM_CAP`. So this is a bit of a foggy thing here. As the `MAXIMUM_CAP` issue and the `INSTALLED_CAP` issue is not fixed, I would leave this PR as it, but mention it somewhere in the checklist there (propably this will result in no loop over the groups anymore).